### ISSUE TITLE
Expose CachedTempBuffer tags to DataBox

### ISSUE THREAD

**Issue by nilsvu** [sxs-collaboration/spectre#4649]
CachedTempBuffer is a way to compute variables lazily, and in contiguous memory (it uses a `Variables` for storage internally). It would be useful to expose the individual tags in the CachedTempBuffer to the DataBox like we do for `Variables`. Then, a CachedTempBuffer is an alternative to the compute item infrastructure for tensors.

**Comment by kidder** [comment]
why would you put a temp buffer into a DataBox?  that defeats the whole purpose of a temp buffer....

**Comment by nilsvu** [comment]
True, it's not "temp". But it's an alternative to writing a compute tag for every single pointwise function and adding them all to the DataBox. With the CachedTempBuffer we avoid compile-time expensive DataBox dependency resolution, the many individual memory allocations for all tensors, and the pattern of doing "tag programming".

**Comment by wthrowe** [comment]
I can't think of a technical issue with this, but I'm not convinced it gains us much.  The dependency resolution only simplifies a little, since you still need (autogenerated) compute tags for all the tensors in the buffer.  Compute-item allocations are only done once per regrid, so unless you're bouncing back and forth to subcell or something they should be negligible.

You do save writing the compute-tag definitions, but instead you have to write the computer for the buffer, which I suspect is more boilerplate than the tags.

**Comment by kidder** [comment]
I don't understand what you want @nilsvu. What is the lifetime of what you want to put into the CachedTempBuffer? 

If you want it to persist in the DataBox, why not just use a Variables?  The memory usage is the same

If you don't need it to persist in the DataBox, there is the ObservationBox which combines a DataBox with a list of compute tags avoiding the compile-time dependencies.   I agree with @wthrowe that there doesn't seem to be much difference between writing compute-tag definitions and temp-buffer computers.
